In [7]:
import scipy.stats as scst
import numpy as np
import pandas as pd

# Q1

In [8]:
data = {'A': [173, 150], 'B': [125, 73]}
df = pd.DataFrame(data, columns =['A', 'B'])
prob = df['A']/(df['B'] + df['A']) 
odds = prob / (1 - prob)
OR = odds[0] / odds[1]
OR

0.6735466666666664

In [ ]:
# conf int for OR

In [13]:
obs = np.array([[173, 125], [150, 73]])
scst.chi2_contingency(obs, correction = True)

(4.210350238164615,
 0.040178040571531876,
 1,
 array([[184.74856046, 113.25143954],
        [138.25143954,  84.74856046]]))

### Chi-square test is chosen as we want to test if age is dependent on political association, which is 2 categorial variable. The test statistic value is 4.21035, the p-value is 0.04, and the conclusion is that the null hypothesis is rejected, which means the two variables are dependent.

# Q2

# T_T no time